addon/mixins/limited-route.js

```diff
@@ -61,4 +68,64 @@ export default Ember.Mixin.create({
     @default null
   */
   filter: null,
+
+  /**
+    Return predicate to filter through.
+
+    @example
+      ```javascript
+      // app/routes/example.js
+      ...
+      predicateForFilter(filter) {
+        switch (filter.type) {
+          case 'ember-flexberry-dummy-gender':
+            return new SimplePredicate(filter.name, 'eq', `cast(${filter.pattern},Edm.Enums)`);
```

Comment:
This is a bad exaple for application developers :) `Cast` is OData-specific function so it would not work in offline mode for example. It's better to specify some another value here (and in same cases for other examples).